crates/noseyparker-cli/src/cmd_github.rs

```diff
@@ -13,13 +17,24 @@ pub fn run(global_args: &GlobalArgs, args: &GitHubArgs) -> Result<()> {
 }
 
 fn list_repos(_global_args: &GlobalArgs, args: &GitHubReposListArgs, api_url: Url) -> Result<()> {
-    if args.repo_specifiers.is_empty() {
+    if args.repo_specifiers.is_empty() && !args.repo_specifiers.scan_github_enterprise_instance {
         bail!("No repositories specified");
     }
+    if let Some(host) = api_url.host_str() {
+        if host == "api.github.com" && args.repo_specifiers.scan_github_enterprise_instance {
+            let mut cmd = CommandLineArgs::command();
+            let err = cmd.error(
+                ErrorKind::MissingRequiredArgument,
+                "The custom GitHub API URL was not specified",
+            );
+            err.exit();
+        }
+    }
```

Comment:
Since scanning all public GitHub organizations will be infeasible, let's keep this error-checking behavior: Nosey Parker will refuse to run with the `--all-organizations` unless a custom `--github-api-url` argument is given.

Reply:
Absolutely! At first, I forgot to add this check, and the tests would time out as they were trying to scan all public GitHub organizations...